toolkit/quickrebuild_autoconfig.mk

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+# Contains:
+#	- build optimization flag auto configuration
+
+######## QUICKREBUILD AUTO CONFIGURE ########
+
+# The QUICKREBUILD* flags are special flags that will try to build the toolchain and packages as quickly as possible. They will
+# automatically set REBUILD_TOOLS, USE_CCACHE, REBUILD_TOOLCHAIN, DELTA_BUILD, INCREMENTAL_TOOLCHAIN, and ALLOW_TOOLCHAIN_DOWNLOAD_FAIL to 'y'.
+QUICKREBUILD           ?= n
+QUICKREBUILD_TOOLCHAIN ?= n
+QUICKREBUILD_PACKAGES  ?= n
+
+ifeq ($(QUICKREBUILD),y)
+QUICKREBUILD_TOOLCHAIN = y
+QUICKREBUILD_PACKAGES  = y
+endif
+
+######## QUICKREBUILD TOOLCHAIN ########
+
+ifeq ($(QUICKREBUILD_TOOLCHAIN),y)
+# If any of these are already set to 'n', report an error
+ifneq ($(filter n,$(REBUILD_TOOLCHAIN)),)
+$(error QUICKREBUILD cannot be used with REBUILD_TOOLCHAIN explicitly set to 'n')
+endif
+ifneq ($(filter n,$(INCREMENTAL_TOOLCHAIN)),)
+$(error QUICKREBUILD cannot be used with INCREMENTAL_TOOLCHAIN explicitly set to 'n')
+endif
+ifneq ($(filter n,$(ALLOW_TOOLCHAIN_DOWNLOAD_FAIL)),)
+$(error QUICKREBUILD cannot be used with ALLOW_TOOLCHAIN_DOWNLOAD_FAIL explicitly set to 'n')
+endif
+
+# Don't care if REBUILD_TOOLS is set or not, doesn't matter to the quickbuild.
```

Comment:
If it doesn't matter, why not remove setting it from this file completely?

Reply:
See above comment as well, but I'd like these flags to be a new user-friendly way to run the tools. Going forward I think the REBUILD_TOOLS flag should just be defaulted to 'y' anyways.